xarray/tests/test_plot.py

```diff
@@ -85,52 +85,54 @@ def test_all_figures_closed():
 
 @pytest.mark.flaky
 @pytest.mark.skip(reason="maybe flaky")
-def text_in_fig():
+def text_in_fig() -> set[str]:
     """
     Return the set of all text in the figure
     """
-    return {t.get_text() for t in plt.gcf().findobj(mpl.text.Text)}
+    return {t.get_text() for t in plt.gcf().findobj(mpl.text.Text)}  # type: ignore[attr-defined] # mpl error?
 
 
-def find_possible_colorbars():
+def find_possible_colorbars() -> list[mpl.collections.QuadMesh]:
     # nb. this function also matches meshes from pcolormesh
-    return plt.gcf().findobj(mpl.collections.QuadMesh)
+    return plt.gcf().findobj(mpl.collections.QuadMesh)  # type: ignore[return-value] # mpl error?
```

Comment:
I believe the typing for `findobj` is wrong. Looks to me it always returns the same type as the input.
https://github.com/matplotlib/matplotlib/blob/78b6bdc04f89dafe4d157855a9023826cab8a0fd/lib/matplotlib/artist.pyi#L131-L135
Probably should be something like `TypeVar("T_Artist", bound=Artist)`.

Reply:
cc @ksunden if you're still interested.

Reply:
See matplotlib/matplotlib#28031

Reply:
Nice, thank you @ksunden !